### PR TITLE
Changed EMBEDLY_KEY to WAGTAILEMBEDS_EMBEDLY_KEY

### PR DESCRIPTION
**Pull request by anurag-ks**
https://github.com/torchbox/wagtail/issues/1757
I was introduced to wagtail few hours ago, so I am not sure if that's all the change that is required.

/wagtail/wagtail/wagtailembeds/embeds.py 
Line Number: 34
        key = settings.EMBEDLY_KEY
Line Number: 118
    if hasattr(settings, 'EMBEDLY_KEY'):

### CODE REVIEW

**Comment by kaedroho** [comment]
Thanks!

We need to also update Wagtail to use the new setting (with backwards compatibility + deprecation warning for the old setting): https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailembeds/embeds.py#L34

**Comment by anurag-ks** [comment]
Okay on it

**Comment by anurag-ks** [comment]
@kaedroho For the warning message, should I just print it to the console or do you guys use some logger ?

**Comment by kaedroho** [comment]
We raise a warning using one of the classes in this file: https://github.com/torchbox/wagtail/blob/master/wagtail/utils/deprecation.py

Here's an example: https://github.com/torchbox/wagtail/blob/490ecff007ef9828947e0718a5b76559e90369e4/wagtail/wagtailcore/models.py#L272-L274

We haven't yet added a ``RemovedInWagtail14Warning`` (marking features to be removed in Wagtail 1.4). Feel free to add that class.

**Comment by kaedroho** [comment]
Yep, line 118 needs to also check for the new setting:
```python
if hasattr(settings, 'WAGTAILEMBEDS_EMBEDLY_KEY') or if hasattr(settings, 'EMBEDLY_KEY'):
```

**Comment by anurag-ks** [comment]
@kaedroho could you merge this now ?

File: wagtail/wagtailembeds/embeds.py [hunk 1]
```diff
@@ -31,7 +33,13 @@ def embedly(url, max_width=None, key=None):
 
     # Get embedly key
     if key is None:
-        key = settings.EMBEDLY_KEY
+        try:
+            key = settings.WAGTAILEMBEDS_EMBEDLY_KEY
+        except AttributeError:
+            warnings.warn(
+                "use WAGTAILEMBEDS_EMBEDLY_KEY instead",
```
Comment (kaedroho) [inline wagtail/wagtailembeds/embeds.py]: We should give this message some more context:
"EMBEDLY_KEY is now deprecated. Use WAGTAILEMBEDS_EMBEDLY_KEY instead"

File: wagtail/wagtailembeds/embeds.py [hunk 2]
```diff
@@ -11,6 +12,7 @@
 
 from wagtail.wagtailembeds.oembed_providers import get_oembed_provider
 from wagtail.wagtailembeds.models import Embed
+from wagtail.utils.deprecation import RemovedInWagtail13Warning
```
Comment (gasman) [inline wagtail/wagtailembeds/embeds.py]: We should use `RemovedInWagtail14Warning` here instead - we want to give two versions' notice before this is removed.

(`RemovedInWagtail14Warning` has just been added, in 9197bf1259df43a9564495a6b41fb733b025d804)

**Comment by anurag-ks** [comment]
Resending again

**Comment by anurag-ks** [comment]
@kaedroho done.

**Pull request merged**